linode/manager
# fix: [M3-8492] - Create Linode Firewall warning not appearing

## Description 📝
Fixes a bug causing the Firewall warning to not appear after the user assigns and then clears the firewall select during the create flow.

## Changes  🔄
- Update the `firewall_id` interface in `CreateLinodePayload` to not accept null values (i.e., `number | undefined`)
- Refactor `SelectFirewallPanel` and associated types in Create v1 to use `undefined` instead of `-1` when the firewall field is cleared
- Remove a redundant `useEffect`

## Target release date 🗓️
9/3

## Preview 📷
| Before  | After   |
| ------- | ------- |
| **Create v2** |
|  <video src="https://github.com/user-attachments/assets/747aaece-561c-447c-a43d-1b691e964c9c" /> | <video src="https://github.com/user-attachments/assets/f52ccd5a-dd9a-4b25-abf9-64f121b0b160" /> |
| **Create v1** |
| <video src="https://github.com/user-attachments/assets/5e3a402a-3edb-4fd8-8a74-4c871cf720f1" />| <video src="https://github.com/user-attachments/assets/34de2782-4111-4363-b1a5-7648ff9a0265" /> |

## How to test 🧪

### Prerequisites
Enable Secure VM notices by adding `secure_vm_notices: 'always'` in the [preference editor](http://localhost:3000/profile/settings?preferenceEditor=true).

### Reproduction steps
- Select and then clear the firewall field in the Linode Create flows (v1 and v2)
- Click the "Create Linode" button
- Observe that the Firewall authorization warning banner does not appear
    - In Create v1, the form is submitted and the Linode is created
    - In Create v2, the Create Linode button becomes disabled

### Verification steps
- Follow the reproduction steps
- Verify that the Firewall authorization banner always appears after clicking the Create Linode button when no firewall is assigned
